src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -1580,8 +1614,8 @@ private JsonRpcMessage HandleInvocationTaskOfTResult(JsonRpcRequest request, Tas
 
             if (message is JsonRpcResult resultMessage)
             {
-                // This method should only be called for methods that declare to return Task<T> or a derived type.
-                Assumes.True(TryGetTaskOfTType(t.GetType().GetTypeInfo(), out TypeInfo taskOfTTypeInfo));
+                // This method should only be called for methods that declare to return Task<T> or a derived type or ValueTask<T>.
```

Comment:
nit: "// This method should only be called for methods that declare to return Task<T>, a derived type, or ValueTask<T>."

Reply:
Fixed (not quite that way, but I hope it fits what you were going for).